packages/firestore/rollup.config.js

```diff
@@ -15,27 +15,72 @@
  * limitations under the License.
  */
 
+import * as path from 'path';
+
 import json from 'rollup-plugin-json';
 import typescriptPlugin from 'rollup-plugin-typescript2';
 import replace from 'rollup-plugin-replace';
 import copy from 'rollup-plugin-copy-assets';
 import typescript from 'typescript';
+import { terser } from 'rollup-plugin-terser';
+
+import { renameInternals } from './scripts/rename-internals';
+import { extractPublicIdentifiers } from './scripts/extract-api';
 import pkg from './package.json';
+import { externs } from './externs.json';
 
 const deps = Object.keys(
   Object.assign({}, pkg.peerDependencies, pkg.dependencies)
 );
 
+// Extract all identifiers used in external APIs (to be used as a blacklist by
+// the SDK minifier).
+const externsPaths = externs.map(p => path.resolve(__dirname, '../../', p));
+const publicIdentifiers = extractPublicIdentifiers(externsPaths);
+const transformers = [
+  service => ({
+    before: [
+      renameInternals(service.getProgram(), {
+        publicIdentifiers,
+        prefix: '__PRIVATE_'
+      })
+    ],
+    after: []
+  })
+];
+
+const terserOptions = {
+  output: {
+    comments: false
+  },
+  mangle: {
+    properties: {
+      regex: /^__PRIVATE_/
+    }
+  }
+};
+
 /**
  * ES5 Builds
  */
 const es5BuildPlugins = [
   typescriptPlugin({
-    typescript
+    typescript,
+    cacheRoot: './.cache/es5/'
```

Comment:
Just curious about the reason for setting custom `cacheRoot` dirs for all the builds in this file.

Reply:
This drove me nuts until I figured it out: The cache for the typescript plugin prevents any rebuild for any typescript file if it hasn't changed - regardless of transformers. So if you have two different builds that each touch the same file, the cache is re-used and the cached result from the first build is used for both builds. I solved this by using a different cache for each transformer setting.